src/rule-unused-fragments.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const utils = require('./utils');
+
+const getGraphQLAST = utils.getGraphQLAST;
+
+function getGraphQLFragmentNames(graphQLAst) {
```

Comment:
I know we use this function in another rule, but would you be able to refactor this to use the [`visit`](https://github.com/graphql/graphql-js/blob/master/src/language/visitor.d.ts#L238-L242) function from the `graphql` package? You can refer to this for an [example](https://github.com/relayjs/eslint-plugin-relay/blob/master/src/rule-compat-uses-vars.js#L17). You can look up the node's ancestors (ie to check if its within an `OperationDefinition` with that pretty easily.